src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -67,6 +80,11 @@ export class VSCodeNotebookController implements Disposable {
     get onDidReceiveMessage() {
         return this.controller.onDidReceiveMessage;
     }
+    @testOnlyMethod()
+    public isAssociatedWithDocument(doc: NotebookDocument) {
+        return this.associatedDocuments.has(doc);
+    }
+    private readonly associatedDocuments = new WeakSet<NotebookDocument>();
```

Comment:
Instead of `controllerMapping`, the controller will have its own property to keep track of whether a document is associated with this controller or not.